packages/types/src/discord.ts

```diff
@@ -2373,27 +2373,70 @@ export interface DiscordApplicationCommandOption {
   description: string
   /** Localization object for the `description` field. Values follow the same restrictions as `description` */
   description_localizations?: Localization | null
-  /** If the parameter is required or optional--default `false` */
+  /**
+   * If the parameter is required or optional. default `false`
+   *
+   * @remarks
+   * Valid in all option types except {@link ApplicationCommandOptionTypes.SubCommand | SubCommand} and {@link ApplicationCommandOptionTypes.SubCommandGroup | SubCommandGroup}
+   */
   required?: boolean
-  /** Choices for the option types `ApplicationCommandOptionTypes.String`, `ApplicationCommandOptionTypes.Integer`, and `ApplicationCommandOptionTypes.Number`, from which the user can choose, max 25 */
+  /**
+   * Choices for the option from which the user can choose, max 25
+   *
+   * @remarks
+   * Only valid in options of type {@link ApplicationCommandOptionTypes.String | String}, {@link ApplicationCommandOptionTypes.Integer | Integer}, or {@link ApplicationCommandOptionTypes.Number | Number}
+   *
+   * If you provide there options, they will be the ONLY accepted values for this option
+   */
   choices?: DiscordApplicationCommandOptionChoice[]
-  /** If the option is a subcommand or subcommand group type, these nested options will be the parameters */
+  /**
+   * If the option is a subcommand or subcommand group type, these nested options will be the parameters
+   *
+   * @remarks
+   * Only valid in option of type {@link ApplicationCommandOptionTypes.SubCommand | SubCommand} or {@link ApplicationCommandOptionTypes.SubCommandGroup | SubCommandGroup}
+   */
   options?: DiscordApplicationCommandOption[]
   /**
    * If autocomplete interactions are enabled for this option.
    *
-   * Only available for `ApplicationCommandOptionTypes.String`, `ApplicationCommandOptionTypes.Integer` and `ApplicationCommandOptionTypes.Number` option types
+   * @remarks
+   * Only valid in options of type {@link ApplicationCommandOptionTypes.String | String}, {@link ApplicationCommandOptionTypes.Integer | Integer}, or {@link ApplicationCommandOptionTypes.Number | Number}
    */
   autocomplete?: boolean
```

Comment:
Should we document that `autocomplete may not be set to true if choices are present.`?

Reference: https://github.com/discord/discord-api-docs/pull/6177/files#diff-eac5037296fa30697ac678095a75beba6d78d1b337263bd2d06d85b147dc616aR72